essencium-backend/src/main/java/de/frachtwerk/essencium/backend/service/AbstractUserService.java

```diff
@@ -386,4 +398,36 @@ public List<SessionToken> getTokens(USER user) {
   public void deleteToken(USER user, @NotNull UUID id) {
     jwtTokenService.deleteToken(user.getUsername(), id);
   }
+
+  @Override
+  protected void deletePreProcessing(@NotNull final ID id) {
+    super.deletePreProcessing(id);
+
+    if (Objects.nonNull(SecurityContextHolder.getContext().getAuthentication())) {
+      var executingUser = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+      if (executingUser instanceof AbstractBaseUser<?>) {
+        boolean doModifyMyself = ((AbstractBaseUser<?>) executingUser).getId() == id;
+        if (doModifyMyself) {
+          throw new NotAllowedException(
+              "You cannot delete yourself. That is to ensure there's at least one ADMIN remaining.");
+        }
+      }
+    }
+  }
+
+  private void abortWhenRemovingAdminRole(ID id, boolean remainAdmin) {
+    if (Objects.isNull(SecurityContextHolder.getContext().getAuthentication())) {
+      return;
+    }
+    var executingUser = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    if (executingUser instanceof AbstractBaseUser<?>) { // OAuth2User rights are not managed by us
+      boolean doModifyMyself = ((AbstractBaseUser<?>) executingUser).getId() == id;
```

Comment:
Why are you retracting the change here? The following code works:
```java
if (Objects.nonNull(SecurityContextHolder.getContext().getAuthentication())) {
      AbstractBaseUser<ID> executingUser =
          (AbstractBaseUser<ID>)
              SecurityContextHolder.getContext().getAuthentication().getPrincipal();
      boolean doModifyMyself = Objects.equals(executingUser.getId(), id);
      if (doModifyMyself) {
        throw new NotAllowedException(
            "You cannot delete yourself. That is to ensure there's at least one ADMIN remaining.");
      }
```

Reply:
Implementing your change in `deletePreProcessing` works, yes, but if I implement the same change in `abortWhenRemovingAdminRole` then I get errors in `AuthenticationControllerIntegrationTest::Oauth` so I guess the change itself is not "sane" and only passes because of limited code coverage